### PR TITLE
[zeroc-ice] Update to 3.7.9 and a few fixes/improvements of the port

### PR DESCRIPTION
**Pull request by bold84**
- With the update to 3.7.9, the slice compiler (Ice Builder) had to be updated to 5.0.9. 
- Out of the box, Ice (at least on Windows) seems to only support static builds for the libraries, but not the executables. Therefore I have added "supports" clauses for the features that are incompatible with static builds. 

If this PR updates an existing port, please uncomment and fill out this checklist:

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/zeroc-ice/portfile.cmake [hunk 1]
```diff
@@ -2,9 +2,9 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO zeroc-ice/ice
-    REF v3.7.7
-    SHA512 73c3a2bb14c9e145383e4026206edd3e03b29c60a33af628611bfdab71d69a3aed108ce4e6cbfd67eb852560110e3495b4bd238c8cdf0de9d1f8e2f1088513ee
-    PATCHES md5i_fix.patch slice2swift.patch mcppd_fix.patch no-werror.patch
+    REF v3.7.9
```
Comment (Adela0814) [inline ports/zeroc-ice/portfile.cmake:5]:
```suggestion
    REF "v${VERSION}"

```

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/zeroc-ice/prepare_for_build.cmake [hunk 2]
```diff
@@ -50,15 +51,16 @@ function(prepare_for_build ICE_SOURCE_DIR)
         )
     endforeach()
 
+	set(ICE_BUILDER_VERSION "5.0.9")
     set(ICE_CPP_MSBUILD_PACKAGES_DIR "${ICE_SOURCE_DIR}/cpp/msbuild/packages")
```
Comment (dg0yt) [inline ports/zeroc-ice/prepare_for_build.cmake:55]: Please fix the indentation.

**Comment by Adela0814** [comment]
Tested usage and all feature on the following triplets:

- x64-windows
- x64-windows-static
- x86-windows

**Review by Adela0814 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**